pkg/compiler/validators/typing.go

```diff
@@ -33,6 +33,20 @@ func (t trivialChecker) CastsFrom(upstreamType *flyte.LiteralType) bool {
 	if isVoid(upstreamType) {
 		return true
 	}
+
+	// If upstream is an enum, it can be consumed as a string downstream
+	if upstreamType.GetEnumType() != nil {
```

Comment:
We should also add a check to the static binding to fail if the string isn't in the list...